pkgs/development/python-modules/ifcopenshell/default.nix

```
@@ -1,67 +1,192 @@
 {
   lib,
   stdenv,
+  testers,
   buildPythonPackage,
+  pythonOlder,
+  python,
+  pytestCheckHook,
+  # fetchers
   fetchFromGitHub,
-  gcc10,
+  # build tools
   cmake,
+  swig,
+  # native dependencies
+  eigen,
   boost179,
+  cgal,
+  gmp,
+  hdf5,
   icu,
-  swig,
-  pcre,
+  libaec,
+  libxml2,
+  mpfr,
+  nlohmann_json,
   opencascade-occt_7_6,
   opencollada,
-  libxml2,
+  pcre,
+  zlib,
+
+  # python deps
+  ## tools
+  setuptools,
+  build,
+  pytest,
+  ## dependencies
+  isodate,
+  lark,
+  numpy,
+  python-dateutil,
+  shapely,
+  typing-extensions,
+  ## additional deps for tests
+  lxml,
+  mathutils,
+  networkx,
+  tabulate,
+  xmlschema,
+  xsdata,
 }:
 let
   opencascade-occt = opencascade-occt_7_6;
 in
 buildPythonPackage rec {
   pname = "ifcopenshell";
-  version = "240611";
+  version = "0.7.10";
   format = "other";
+  disabled = pythonOlder "3.9";
```

```suggestion
  pyproject = false;
```
we don't need to add disabled checks for old python versions